spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkWriteConf.java

```diff
@@ -179,7 +179,7 @@ public boolean fanoutWriterEnabled() {
         .booleanConf()
         .option(SparkWriteOptions.FANOUT_ENABLED)
         .tableProperty(TableProperties.SPARK_WRITE_PARTITIONED_FANOUT_ENABLED)
-        .defaultValue(TableProperties.SPARK_WRITE_PARTITIONED_FANOUT_ENABLED_DEFAULT)
+        .defaultValue(true)
```

Comment:
The proposed approach simply enables fanout writers by default. Our distribution and ordering code checks if the table is unsorted and whether fanout writers are enabled to skip a local sort. Previously, we only skipped local sorts if fanout writers were enabled explicitly.

Even though we enable fanout writers by default, it does not mean they will be used. Our writers later check if the input is ordered. If yes, we will still use clustered writers.

Reply:
Should we just change the `TableProperties` default to `true` instead of hard-coding a different value here?

Reply:
Actually, see my comment below about making this optional and inferring only when it is not explicitly set on the table or in write options. I think that's much safer.

Reply:
I cannot change the value of the property because other Spark versions depend on it. I did change the logic to infer only if not set, however.